src/handle_init_contract.c

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+ *
+ * ██╗  ██╗██╗██╗     ███╗   ██╗
+ * ██║ ██╔╝██║██║     ████╗  ██║
+ * █████╔╝ ██║██║     ██╔██╗ ██║
+ * ██╔═██╗ ██║██║     ██║╚██╗██║
+ * ██║  ██╗██║███████╗██║ ╚████║
+ * ╚═╝  ╚═╝╚═╝╚══════╝╚═╝  ╚═══╝
+ *
+ * Kiln Ethereum Ledger App
+ * (c) 2022-2024 Kiln
+ *
+ * contact@kiln.fi
+ ********************************************************************************/
+
+#include "kiln_plugin.h"
+#include "plugin_utils.h"
+
+void handle_init_contract(ethPluginInitContract_t *msg) {
+    if (msg->interfaceVersion != ETH_PLUGIN_INTERFACE_VERSION_LATEST) {
+        msg->result = ETH_PLUGIN_RESULT_UNAVAILABLE;
+        return;
+    }
+
+    if (msg->pluginContextLength < sizeof(context_t)) {
+        PRINTF("Plugin parameters structure is bigger than allowed size\n");
+        msg->result = ETH_PLUGIN_RESULT_ERROR;
+        return;
+    }
+
+    context_t *context = (context_t *) msg->pluginContext;
+
+    memset(context, 0, sizeof(*context));
+
+    size_t index;
+    if (!find_selector(U4BE(msg->selector, 0), KILN_SELECTORS, NUM_SELECTORS, &index)) {
+        PRINTF("Error: selector not found!\n");
+        msg->result = ETH_PLUGIN_RESULT_UNAVAILABLE;
+        return;
+    }
+    context->selectorIndex = index;
+    // check for overflow
+    if ((size_t) context->selectorIndex != index) {
+        PRINTF("Error: overflow detected on selector index!\n");
+        msg->result = ETH_PLUGIN_RESULT_ERROR;
+        return;
+    }
+
+    msg->result = ETH_PLUGIN_RESULT_OK;
+
+    switch (context->selectorIndex) {
+        case KILN_V1_DEPOSIT:
```

Comment:
super-nit: maybe group only by product, and remove the extra lines within v1 selectors, to keep the v1/v2/LR breakdown